src/app/frontend/deploy/deploy_controller.js

```diff
@@ -112,6 +117,34 @@ export default class DeployController {
   }
 
   /**
+   * Displays new namespace creation dialog.
+   *
+   * @param {!angular.Scope.Event} event
+   */
+  showNamespaceDialog(event) {
+    this.mdDialog_.show({
+                    controller: NamespaceDialogController,
+                    controllerAs: 'ctrl',
+                    clickOutsideToClose: true,
+                    targetEvent: event,
+                    templateUrl: '/deploy/namespace_dialog.html',
+                    locals: {
+                      namespaces: this.namespaces,
+                    },
+                  })
+        .then(
+            function(answer) {
```

Comment:
What is the type of `answer`? This is non trivial, can you annotate the function?